src/librustdoc/html/render.rs

```diff
@@ -2285,15 +2288,13 @@ fn short_stability(item: &clean::Item, cx: &Context) -> Vec<String> {
         ));
     }
 
-    if let Some(stab) = item.stability.as_ref().filter(|stab| stab.level == stability::Unstable) {
-        let is_rustc_private = stab.feature.as_deref() == Some("rustc_private");
-
-        let mut message = if is_rustc_private {
-            "<span class='emoji'>⚙️</span> This is an internal compiler API."
-        } else {
-            "<span class='emoji'>🔬</span> This is a nightly-only experimental API."
-        }
-        .to_owned();
+    // Render unstable items. But don't render "rustc_private" crates (internal compiler crates).
+    // Those crates are permanently unstable so it makes no sense to render "unstable" everywhere.
+    if let Some(stab) = item.stability.as_ref().filter(|stab| {
+        stab.level == stability::Unstable && stab.feature.as_deref() != Some("rustc_private")
```

Comment:
nit: we should be comparing against `rustc_span::sym::rustc_private` to avoid hitting the string interner lock here.

Reply:
I would love to. But [`stab.feature`](https://doc.rust-lang.org/nightly/nightly-rustc/rustdoc/clean/types/struct.Stability.html) is `Option<String>`. I'm planning for merging in https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/Merge.20Stability.20in.20rustc_attr.20and.20rustdoc .

Reply:
Oh right this is rustdoc-cleaned stability. Fine as is then.